server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -317,8 +324,16 @@ public void childEvent(
     catch (Exception ex) {
       throw ex;
     }
-    Assert.assertTrue(assignSegmentLatch.await(4, TimeUnit.SECONDS));
+
+    Assert.assertTrue(assignSegmentLatch.await(2, TimeUnit.SECONDS));
+    Assert.assertEquals(ImmutableMap.of(dataSource, 100.0), coordinator.getLoadStatus());
     curator.delete().guaranteed().forPath(ZKPaths.makePath(LOADPATH, dataSegment.getIdentifier()));
+    // Wait for coordinator thread to run so that replication status is updated
+    // Coordination period is 100ms so wait of 150ms should be sufficient
+    Thread.sleep(150);
```

Comment:
since the coordinator period and this sleep time are related then may be it makes sense to make coordinator an instance variable and do a Thread.sleep(coordinator_period + 50) here.